src/devices/plic.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#include <assert.h>
+#include <stdlib.h>
+
+#include "../riscv.h"
+#include "../riscv_private.h"
+#include "plic.h"
+
+void plic_update_interrupts(plic_t *plic)
+{
+    riscv_t *rv = (riscv_t *) plic->rv;
+
+    /* Update pending interrupts */
+    plic->ip |= plic->active & ~plic->masked;
+    plic->masked |= plic->active;
+    /* Send interrupt to target */
+    if (plic->ip & plic->ie)
+        rv->csr_sip |= SIP_SEIP;
+    else
+        rv->csr_sip &= ~SIP_SEIP;
+}
+
+uint32_t plic_read(plic_t *plic, const uint32_t addr)
+{
+    /* no priority support: source priority hardwired to 1 */
+    if (1 <= addr && addr <= 31)
+        return 0;
```

Comment:
Read for the non-supported register has already defaulted to 0. We don't need this branch condition here as it should be unlikely to hit.